### ISSUE TITLE
Using recorded audio?

### ISSUE THREAD

**Issue by ab36245** [csdcorp/speech_to_text#205]
This may be a slightly odd question, but for testing and other reasons I'd like to be able to use the plugin with recorded voice. I don't know enough about the underlying recognizers (either Android or iOS) to know whether it is possible to feed them audio rather than only use microphone input.

Is anything like this possible?

**Comment by sowens-csd** [comment]
I've looked into this a couple of times and so far I have not found any way to achieve this using the OS speech recognition packages.

**Comment by ab36245** [comment]
Understood, and I agree that there doesn't seem to be a whole lot of information out there. It's a shame because the more I think about this the more useful it seems to be.

This [SO post](https://stackoverflow.com/questions/49443493/swift-how-can-i-convert-saved-audio-file-conversations-to-text) is about 3 years old but suggests something might be possible on iOS?

On Android I can't seem to find any hints that this is possible so far.

**Comment by sowens-csd** [comment]
Yes, iOS is likely feasible. I do have access to the raw audio content on iOS. Obviously given Flutter's cross platform goals it would be better if there was a solution for both.

**Comment by ab36245** [comment]
Oh I agree this is Flutter so it should be cross platform. Just trying to get the ball rolling. From the little I've researched I'm surprised how little control Android APIs provide over speech recognition in general. I'll keep looking though

**Comment by tamoyal** [comment]
@sowens-csd The docs say it is supported on iOS - https://developer.apple.com/documentation/speech#topics

> SFSpeechAudioBufferRecognitionRequest
A request to recognize speech from captured audio content, such as audio from the device's microphone.

**Comment by sowens-csd** [comment]
@tamoyal I agree. The problem is that it isn't possible on Android and I've been resisting a single platform solution so far. Do you disagree?

**Comment by tamoyal** [comment]
@sowens-csd I would think it's best to enable as much as possible regardless of platform (btw isn't the sound level listener Android only)? If I'm developing an app and want it to work on Android and iOS, before Flutter, I would develop two separate codebases and they would have their own limitations. With Flutter, it's not strange to think I still have to consider those limitations but I really want the full power of each device.

Practically in my case, I would love to use this on iOS only. Actually might have to slowly ditch the library if it's not enabled ..but I'm also one person with one use case and I'm guessing you have talked to many.

But yea... philosophically I think letting the consumer decide if they want to have different features per device and not having it be a limitation of the library makes a lot of sense. 

I also don't know how much work it is....

**Comment by sowens-csd** [comment]
Interesting point. I'm also not sure how much work it is but I'll keep it in mind.

**Comment by GiuseppeCrBrandi** [comment]
Come on guys. How can we do speech to text but not pre-recorded audio to text? I'm really having a hard time trying to understand it. It's all bytes in the end. Why not make the necessary adaptations for a very useful feature. We dont want to use the expensive Google Cloud API

**Comment by filip-alexandrov** [comment]
It seems the only feasible option for Android ASR from an audio file is loading your own model. 
Vosk has a good variety of models, but only available on Android: https://github.com/alphacep/vosk-flutter
